### ISSUE TITLE
issue when using make for pi zero w

### ISSUE THREAD

**Issue by keganpowers34** [GaryOderNichts/udpih#4]
make -C /lib/modules/5.15.32+/build M=/home/pi/udpih/linux modules
make[1]: *** /lib/modules/5.15.32+/build: No such file or directory.  Stop.
make: *** [Makefile:11: all] Error 2

**Comment by keganpowers34** [comment]
am i doing something wrong when compiling the kernel?

**Comment by GaryOderNichts** [comment]
Did you install the dependencies?

From the README:
- Install the required dependencies:
```
sudo apt install build-essential raspberrypi-kernel-headers
```

**Comment by keganpowers34** [comment]
Yes I did

**Comment by keganpowers34** [comment]
it might be the os i have on the pi zero what did you use for your pi zero?

**Comment by keganpowers34** [comment]
i have not been able to fix this any ideas?

**Comment by GaryOderNichts** [comment]
Make sure you're on the latest stable Raspberry Pi OS version by running:
```
sudo rpi-update stable
```
Otherwise the kernel headers won't match your currently running kernel version.

**Comment by keganpowers34** [comment]
alright so i did get it working but when plugging in the usb cable for the pi when the wii u menu comes up the recovery menu doesn't popup the pi is powered and the recovery menu is on the root of the sd card and the sd card is fat 32 formated

**Comment by GaryOderNichts** [comment]
Make sure to plug it in once the Wii U logo comes up.
If you already see the Wii U menu it's too late.

**Comment by keganpowers34** [comment]
it did not work i also tried all the usb ports to see if that would make a change

**Comment by GaryOderNichts** [comment]
Make sure no other devices are plugged in and that you're using the front USB ports.
Also check the logs on your Pi using `dmesg`.

**Comment by keganpowers34** [comment]
nothing out of the ordinary happened except for this
`
[  130.768655] udpih: loading out-of-tree module taints kernel.
[  130.769807] udpih_init
[  130.769838] udc-core: couldn't find an available UDC - added [Wii U USB Host Stack exploit] to list of pending drivers
`

**Comment by keganpowers34** [comment]
`[   92.543042] udpih: loading out-of-tree module taints kernel.
[   92.544193] udpih_init
[   92.544226] udc-core: couldn't find an available UDC - added [Wii U USB Host Stack exploit] to list of pending drivers
[  113.610501] dwc2 20980000.usb: new device is high-speed
[  113.670555] dwc2 20980000.usb: new address 1
[  113.688828] IPv6: ADDRCONF(NETDEV_CHANGE): usb0: link becomes ready
[  163.970738] dwc2 20980000.usb: new device is high-speed
[  164.521812] dwc2 20980000.usb: new device is full-speed
[  192.802763] dwc2 20980000.usb: new device is high-speed
[  192.862278] dwc2 20980000.usb: new address 1
`
so it is doing somthing

**Comment by GaryOderNichts** [comment]
> :information_source: To use USB gadgets on the Pi Zero you need to enable the dwc2 module by running the commands below:
> `echo "dtoverlay=dwc2" | sudo tee -a /boot/config.txt`
> `echo "dwc2" | sudo tee -a /etc/modules`
> After running the commands reboot the system.

Did you do this?

**Comment by keganpowers34** [comment]
yes

**Comment by keganpowers34** [comment]
nothing is printing after the line 
`Nov 13 01:19:17 raspberrypi kernel: [  155.720422] IPv6: ADDRCONF(NETDEV_CHANGE): usb0: link becomes ready`

**Comment by keganpowers34** [comment]
ill try the other recovery menu file to see if that works

**Comment by keganpowers34** [comment]
yeah nothing

**Comment by keganpowers34** [comment]
this wii u is not repairible will try on a new one once i get one

**Comment by ProfEibe** [comment]
> Make sure you're on the latest stable Raspberry Pi OS version by running:
> 
> ```
> sudo rpi-update stable
> ```
> 
> Otherwise the kernel headers won't match your currently running kernel version.

you should add this to the readme. It helped me to get it working!

**Comment by partduex** [comment]
I did all the steps mentioned above and still cannot get this to work. I've gotten as far as to the part where I'm supposed to insmod udpih.ko but it gives me an error every time. Oh I also had to use buster os as i wasn't getting as far with bullseye and the echo file and the dwc2 file are 0 bytes is that correct? Oh by the way I'm a complete noob to this any help would be greatly appreciated  . Thank you
